src/core/opamCoreConfig.ml

```diff
@@ -157,11 +159,21 @@ let initk k =
         | true -> Some `Extended
         | false -> None)
     ) in
+  let answer =
+    (* priorities: confirm level > yes > no *)
+    match E.confirmlevel (), E.yes (), E.no () with
+    | Some c, _,  _ -> Some c
+    | _, Some true, _ -> Some `all_yes
+    | _, Some false, _
```

Comment:
```suggestion
    | _, Some false, _ -> None
```
At present, `OPAMYES=false opam ...` still prompts, where this would still answer no.